.github/workflows/CI.yml

```
@@ -172,17 +175,24 @@ jobs:
           name: wheels-sdist
           path: dist
 
-  release:
-    name: Release
+  publish:
+    name: Publish Python distributions to PyPI
     runs-on: ubuntu-latest
-    if: "startsWith(github.ref, 'refs/tags/')"
+    if: github.event_name == 'release' && github.event.action == 'published'
     needs: [linux, musllinux, windows, macos, sdist]
+    environment:
+      name: pypi
+      url: https://pypi.org/p/rfc3161-client
+    permissions:
+      # Used to sign the release's artifacts with sigstore-python.
+      # Used to publish to PyPI with Trusted Publishing.
+      id-token: write
     steps:
       - uses: actions/download-artifact@v4
-      - name: Publish to PyPI
-        uses: PyO3/maturin-action@v1
-        env:
-          MATURIN_PYPI_TOKEN: ${{ secrets.PYPI_API_TOKEN }}
         with:
-          command: upload
-          args: --non-interactive --skip-existing wheels-*/*
+          path: dist/
```

Because we're downloading artifacts without specifying their `name`, they will all be downloaded to their own folder, i.e: `dist/wheels-sdist/*.tar.gz`, `dist/wheels-macos-aarch64/*.whl`, etc.

Can be fixed by adding this 

```suggestion
          path: dist/
          merge-multiple: true
```

([docs](https://github.com/actions/download-artifact?tab=readme-ov-file#download-all-artifacts))